selfdrive/car/tesla/carstate.py

```diff
@@ -29,78 +28,80 @@ def update(self, cp, cp_cam):
 
     # Gear
-    ret.gearShifter = GEAR_MAP[self.can_define.dv["DI_torque2"]["DI_gear"].get(int(cp.vl["DI_torque2"]["DI_gear"]), "DI_GEAR_INVALID")]
+    ret.gearShifter = GEAR_MAP[self.can_define.dv["DI_systemStatus"]["DI_gear"].get(int(cp.vl["DI_systemStatus"]["DI_gear"]), "DI_GEAR_INVALID")]
 
     # Buttons
-    buttonEvents = []
+    button_events = []
     for button in BUTTONS:
-      state = (cp.vl[button.can_addr][button.can_msg] in button.values)
+      state = (cp_adas.vl[button.can_addr][button.can_msg] in button.values)
       if self.button_states[button.event_type] != state:
         event = car.CarState.ButtonEvent.new_message()
         event.type = button.event_type
         event.pressed = state
-        buttonEvents.append(event)
+        button_events.append(event)
       self.button_states[button.event_type] = state
-    ret.buttonEvents = buttonEvents
+    ret.buttonEvents = button_events
 
     # Doors
-    ret.doorOpen = any((self.can_define.dv["GTW_carState"][door].get(int(cp.vl["GTW_carState"][door]), "OPEN") == "OPEN") for door in DOORS)
+    ret.doorOpen = any([cp_adas.vl["VCLEFT_doorStatus"]["VCLEFT_frontLatchSwitch"] != 1,
+                        cp_adas.vl["VCLEFT_doorStatus"]["VCLEFT_rearLatchSwitch"] != 1,
+                        cp_adas.vl["VCRIGHT_doorStatus"]["VCRIGHT_frontLatchSwitch"] != 1,
+                        cp_adas.vl["VCRIGHT_doorStatus"]["VCRIGHT_rearLatchSwitch"] != 1,
+                        cp_adas.vl["VCRIGHT_doorStatus"]["VCRIGHT_trunkLatchStatus"] != 2])
 
     # Blinkers
-    ret.leftBlinker = (cp.vl["GTW_carState"]["BC_indicatorLStatus"] == 1)
-    ret.rightBlinker = (cp.vl["GTW_carState"]["BC_indicatorRStatus"] == 1)
+    # maybe use DAS_turnIndicatorRequestReason
+    ret.leftBlinker = (cp_adas.vl["ID3F5VCFRONT_lighting"]["VCFRONT_indicatorLeftRequest"] != 0)
+    ret.rightBlinker = (cp_adas.vl["ID3F5VCFRONT_lighting"]["VCFRONT_indicatorRightRequest"] != 0)
 
     # Seatbelt
-    if self.CP.carFingerprint == CAR.TESLA_MODELS_RAVEN:
-      ret.seatbeltUnlatched = (cp.vl["DriverSeat"]["buckleStatus"] != 1)
-    else:
-      ret.seatbeltUnlatched = (cp.vl["SDM1"]["SDM_bcklDrivStatus"] != 1)
+    ret.seatbeltUnlatched = cp.vl["DriverSeat"]["buckleStatus"] != 1
 
-    # TODO: blindspot
+    # Blindspot
+    ret.leftBlindspot = cp_cam.vl["DAS_status"]["DAS_blindSpotRearLeft"] != 0
+    ret.rightBlindspot = cp_cam.vl["DAS_status"]["DAS_blindSpotRearRight"] != 0
 
     # AEB
     ret.stockAeb = (cp_cam.vl["DAS_control"]["DAS_aebEvent"] == 1)
 
     # Messages needed by carcontroller
-    self.msg_stw_actn_req = copy.copy(cp.vl["STW_ACTN_RQ"])
-    self.acc_state = cp_cam.vl["DAS_control"]["DAS_accState"]
-    self.das_control_counters.extend(cp_cam.vl_all["DAS_control"]["DAS_controlCounter"])
+    self.sccm_right_stalk = copy.copy(cp_adas.vl["SCCM_rightStalk"])
+    self.das_control = copy.copy(cp_cam.vl["DAS_control"])
```

Comment:
no need to copy the whole message, let's save just the counter val which is the only one used?